src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,196 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY;
+    private static final int SOFT_MAX_ARRAY_LENGTH;
+    private static final Object[] DEFAULT_CAPACITY_EMPTY_ELEMENT_DATA;
```

Comment:
This constant is definitely redundant and could be removed and replaced

Reply:
Can I leave these constants? I need them to check whether the new array does not exceed the maximum possible and also in order not to create a new object of an empty array every time. I tried make it without constants, but I also need it and hardkoded it directly in methods, so I think its no sense. Can i leave it?

Reply:
> This constant is definitely redundant and could be removed and replaced

Can I leave these constants? I need them to check whether the new array does not exceed the maximum possible and also in order not to create a new object of an empty array every time. I tried make it without constants, but I also need it and hardkoded it directly in methods, so I think its no sense. Can i leave it?